spec/rails_helper.rb

```diff
@@ -57,12 +57,31 @@ def call(env)
   end
 end
 
+class PlaywrightLogger
```

Comment:
playwright doesnt have a browser.logs API afaict. So we just listen to the events and build our own array of log events.